keyboards/viktus/sp_mini/keymaps/default/keymap.c

```diff
@@ -0,0 +1,144 @@
+  ),
+
+  [_FN2] = LAYOUT_all(
+    _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,
+    _______,   KC_TILD,   KC_EXLM,   KC_AT,     KC_HASH,   KC_DLR,    KC_PERC,   KC_CIRC,   KC_AMPR,   KC_ASTR,   KC_LPRN,   KC_RPRN,   KC_UNDS,   KC_PLUS,   _______,
+    _______,   _______,   _______,   _______,   KC_INS,    KC_PGUP,   KC_HOME,   _______,   _______,   _______,   _______,   KC_COLN,              _______,   _______,
+    _______,   _______,   _______,   _______,   KC_DEL,    KC_PGDN,   KC_END,               _______,   _______,   _______,   _______,   _______,   KC_UP,
+    _______,   _______,   _______,   _______,   _______,   KC_DEL,                                     _______,   _______,   _______,   KC_LEFT,   KC_DOWN,   KC_RGHT
+  ),
+
+  [_FN3] = LAYOUT_all(
+    _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   KC_F24,    _______,   _______,
+    MO(1),     KC_ESC,    KC_Q,      KC_W,      KC_E,      KC_R,      KC_T,      KC_Y,      KC_U,      KC_I,      KC_O,      KC_P,      KC_DEL,    KC_BSPC,   _______,
+    MO(2),     KC_TAB,    KC_A,      KC_S,      KC_D,      KC_F,      KC_G,      KC_H,      KC_J,      KC_K,      KC_L,      KC_QUOT,              KC_ENT,    _______,
+    _______,   KC_LSFT,   KC_Z,      KC_X,      KC_C,      KC_V,      KC_B,                 KC_N,      KC_M,      KC_COMM,   KC_DOT,    KC_RSFT,   KC_UP,
+    _______,   KC_LCTL,   KC_LALT,   KC_LGUI,   KC_FN1,    SPFN1,                                      BSFN2,     KC_RALT,   KC_FN2,    KC_LEFT,   KC_DOWN,   KC_RGHT
+  )
+
+};
+
+void keyboard_pre_init_user(void) {
+    setPinOutput(F5);  // initialize F5 for LED
+    setPinOutput(F6);  // initialize F6 for LED
+    setPinOutput(F7);  // initialize F7 for LED
+
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    writePinLow(F5);
+    writePinLow(F6);
+    writePinLow(F7);
+    switch (get_highest_layer(state)) {
+        case _FN1:
+            writePinHigh(F5);
+            break;
+        case _FN2:
+            writePinHigh(F6);
+            break;
+        case _FN3:  // replace 'XXXX' with the layer or function name
+            writePinHigh(F7);
+            break;
+        case KC_F24:
+            writePinHigh(F7);
+            writePinHigh(F5);
+            writePinHigh(F6);
+            break;
+        }
+    return state;
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_PGDN);
+        } else {
+            tap_code(KC_PGUP);
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+        tap_code(KC_VOLU);
+    } else {
+        tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```

Comment:
Ideally, this should be in the keyboard's c file, so that default behavior can be set without needing it defined in the keymap. 
```suggestion
```